maas/resource_maas_network_interface_link.go

```diff
@@ -187,14 +187,17 @@ func getNetworkInterfaceLinkParams(d *schema.ResourceData, subnetID int) *entity
 	}
 }
 
-func createNetworkInterfaceLink(client *client.Client, machineSystemID string, networkInterfaceID int, params *entity.NetworkInterfaceLinkParams) (*entity.NetworkInterfaceLink, error) {
+func createNetworkInterfaceLink(client *client.Client, machineSystemID string, networkInterface *entity.NetworkInterface, params *entity.NetworkInterfaceLinkParams) (*entity.NetworkInterfaceLink, error) {
 	// Clear existing links
-	_, err := client.NetworkInterface.Disconnect(machineSystemID, networkInterfaceID)
-	if err != nil {
-		return nil, err
+	if networkInterface.Type != "vlan" {
```

Comment:
Let's add a comment mentioning why we need to exclude vlan type interfaces from Disconnect.